confmap/provider/eecprovider/provider.go

```diff
@@ -0,0 +1,208 @@
+// Copyright The OpenTelemetry Authors
+// Copyright Dynatrace LLC
+// SPDX-License-Identifier: Apache-2.0
+
+// This file was copied from the configurablehttpprovider package and modified
+// to support configuration options through query parameter-encoded URL fragments.
+// Additionally, support for two schemes (HTTP/HTTPS) was dropped in favor of
+// configuring TLS through an option on the provider.
+
+package eecprovider // import "github.com/Dynatrace/dynatrace-otel-collector/confmap/provider/eecprovider"
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+type SchemeType string
+
+const (
+	EECScheme SchemeType = "eec"
+)
+
+const ApiTokenHeader = "Api-Token"
+
+type provider struct {
+	caCertPath         string // Used for tests
+	insecureSkipVerify bool   // Used for tests
```

Comment:
In general I would agree. These were taken from the upstream code, and I didn't want to do a major refactor when I hope that we should be able to use the upstream Provider in the future. I think the lack of a really solid way to mock in Go also contributes to this being more of a valid approach, though I haven't explored alternatives for this case in particular.

Reply:
>What mocking options do you miss in go?

Mostly the ability to override concrete types. Interfaces would be able to achieve this.

I also agree that we shouldn't have anything testing-specific in production code if possible. If we decide to keep this Provider longer-term instead of switching to the upstream Providers, we can revisit this and refactor it to our standards. Right now I expect it will be removed within a few months.